src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -35,6 +37,69 @@ private BoundExpression RewriteInterpolatedStringConversion(BoundConversion conv
             return result;
         }
 
+        /// <summary>
+        /// Rewrites the given interpolated string to the set of builder creation and Append calls, returning an array builder of the append calls and the result
+        /// local temp.
+        /// </summary>
+        /// <remarks>Caller is responsible for freeing the ArrayBuilder</remarks>
+        private (ArrayBuilder<BoundExpression> BuilderPatternExpressions, BoundLocal Result) RewriteToInterpolatedStringBuilderPattern(BoundInterpolatedString node)
+        {
+            Debug.Assert(node.InterpolationData is { Construction: not null });
+            Debug.Assert(node.Parts.All(static p => p is BoundCall or BoundStringInsert { Value: BoundCall }));
+            var data = node.InterpolationData.Value;
+            var builderTempSymbol = _factory.InterpolatedStringBuilderLocal(data.BuilderType, data.ScopeOfContainingExpression, node.Syntax);
+            var builderTemp = _factory.Local(builderTempSymbol);
+
+            // PROTOTYPE(interp-string): Support optional out param for whether the builder was created successfully and passing in other required args
+            // var builder = Construction(baseStringLength, numFormatHoles);
+            var builderConstruction = _factory.AssignmentExpression(builderTemp, (BoundExpression)VisitCall(data.Construction));
+
+            var usesBoolReturn = data.UsesBoolReturns;
+            var builderPatternExpressions = ArrayBuilder<BoundExpression>.GetInstance(usesBoolReturn ? 2 : node.Parts.Length);
+            builderPatternExpressions.Add(builderConstruction);
+            BoundExpression? appendBinaryExpression = null;
+            foreach (var currentPart in node.Parts)
+            {
+                var appendCall = (BoundCall)(currentPart is BoundStringInsert si ? si.Value : currentPart);
+                Debug.Assert(usesBoolReturn == (appendCall.Method.ReturnType.SpecialType == SpecialType.System_Boolean));
+
+                var rewrittenArgs = VisitList(appendCall.Arguments);
```

Comment:
>VisitList [](http://example.com/codeflow?start=36&length=9)

Are we doing this manually only because we need to substitute receiver? Consider updating the original node with new receiver (asserting that we are substituting expected node) and then calling VisitCall

Reply:
Partially that, but also partially because the real receiver was created in lowering and `VisitCall` will visit it.